pkg/metrics/metrics.go

```diff
@@ -97,11 +103,14 @@ func initOperatorService(port int32, portName string) (*v1.Service, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	label := map[string]string{"name": operatorName}
+
 	service := &v1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      operatorName,
```

Comment:
Hmm. So I'm wondering if your original instinct to give the operator developer a chance to tweak the service before submitting it to the cluster was a good one.

As is, if an operator developer wants to expose any other ports via a service, they will either have to create a separate service with a different name, or won't be able to enable metrics.

Another option could be to have `ExposeMetricsPort` take a service as an input parameter, and if its nil initialize it with the metrics port. WDYT?

Reply:
I would agree, my only fear is that it will break the API. Can open an issue to discuss this, as it's not strictly related to this PR anyways?

Reply:
Done https://github.com/operator-framework/operator-sdk/issues/1107